testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/component/SampleWebView.kt

```diff
@@ -0,0 +1,38 @@
+package com.rakuten.tech.mobile.testapp.ui.component
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.webkit.WebResourceRequest
+import android.webkit.WebView
+import android.webkit.WebViewClient
+
+@SuppressLint("SetJavaScriptEnabled")
+class SampleWebView(context: Context, url: String, sampleWebViewClient: WebViewClient): WebView(context) {
+
+    init {
+        settings.javaScriptEnabled = true
+        webViewClient = sampleWebViewClient
+        loadUrl(url)
+    }
+}
+
+class SampleWebViewClient(
+    private val finishCallback: (url: String) -> Unit,
+    private val miniAppUrlSchemes: Array<String>
+): WebViewClient() {
+
+    override fun shouldOverrideUrlLoading(view: WebView, request: WebResourceRequest): Boolean {
+        val url = request.url.toString()
+        if (isMiniAppScheme(url))
```

Comment:
I think this functionality could be built-in to the SDK. For example, there could be a `MiniAppExternalUrlLoader(mniAppId: String)` class with a `shouldOverrideUrlLoading(url: String)` function (this is similar to what the `WebViewAssetLoader` does).

As mentioned in another comment, we can create some utility function inside the SDK for generating the custom schemes when given a Mini App ID. That way we can keep the schemes contained in the SDK and simplify use of the SDK a little bit.